src/Microsoft.Azure.Batch.SoftwareEntitlement.Client.Native/SoftwareEntitlementClient.cpp

```diff
@@ -656,8 +595,97 @@ class Curl
 
         throw Exception(GetErrorMessage(code));
     }
+
+    static std::unique_ptr<Entitlement> GetEntitlement(
+        std::string url,
+        const std::string& entitlement_token,
+        const std::string& requested_entitlement)
+    {
+        Curl curl;
+        curl.Post(url + "softwareEntitlements?api-version=2017-05-01.5.0", entitlement_token, requested_entitlement);
+
+        curl.VerifyIntermediateCertificate(url);
+
+        return curl.GetEntitlement();
+    }
+};
+
+struct WinHttpDeleter
+{
+    void operator()(HINTERNET h)
+    {
+        WinHttpCloseHandle(h);
+    }
 };
 
+typedef std::unique_ptr<void, WinHttpDeleter> WinHttpHandle;
+
+//
+// OpenSSL does not hook into the Windows Automatic Root Certificates Update process.
+// This results in certificate validation failures, so we perform a dummy connection
+// using WinHTTP which will setup the root cert store correctly.
+//
+static void EnsureRootCertsArePopulated(const std::string& url)
```

Comment:
This is already instead an anonymous namespace so doesn't need to be `static` as well.

Reply:
done